TrackingTools/TransientTrack/interface/TransientTrackBuilder.h

```diff
@@ -36,12 +37,35 @@ class TransientTrackBuilder {
     std::vector<reco::TransientTrack> build ( const edm::Handle<edm::View<reco::Track> > & trkColl)  const;
 
     std::vector<reco::TransientTrack> build ( const edm::Handle<reco::TrackCollection> & trkColl,
+					      edm::Handle<edm::ValueMap<float> > trackTimes,
+					      edm::Handle<edm::ValueMap<float> > trackTimeResos )  const;
+    std::vector<reco::TransientTrack> build ( const edm::Handle<reco::GsfTrackCollection> & trkColl,
+					      edm::Handle<edm::ValueMap<float> > trackTimes,
+					      edm::Handle<edm::ValueMap<float> > trackTimeResos )  const;
+    std::vector<reco::TransientTrack> build ( const edm::Handle<edm::View<reco::Track> > & trkColl,
+					      edm::Handle<edm::ValueMap<float> > trackTimes,
+					      edm::Handle<edm::ValueMap<float> > trackTimeResos )  const;
+
+    std::vector<reco::TransientTrack> build ( const edm::Handle<reco::TrackCollection> & trkColl,
 	const reco::BeamSpot & beamSpot) const;
     std::vector<reco::TransientTrack> build ( const edm::Handle<reco::GsfTrackCollection> & trkColl,
 	const reco::BeamSpot & beamSpot)  const;
     std::vector<reco::TransientTrack> build ( const edm::Handle<edm::View<reco::Track> > & trkColl,
 	const reco::BeamSpot & beamSpot)  const;
 
+    std::vector<reco::TransientTrack> build ( const edm::Handle<reco::TrackCollection> & trkColl,
+					      const reco::BeamSpot & beamSpot,
+					      edm::Handle<edm::ValueMap<float> > trackTimes,
```

Comment:
```const &```
also, is it really necessary to pass the handle?
This limits the code to having to have these valueMaps persisted.